src/app/dashboard/components/chart/index.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import {
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+} from 'recharts';
+import styled, { useTheme } from 'styled-components';
+
+import { ThemeType } from 'theme/defaults';
+
+import { StockChartData } from '../../sampleChartData';
+
+function linspace(start: number, stop: number, num: number, endpoint = false) {
+  const div = endpoint ? num - 1 : num;
+  const step = roundTo10((stop - start) / div);
+  return Array.from({ length: num }, (_, i) => start + step * i);
+}
+
+function roundTo10(value: number) {
+  return Math.round(value / 10) * 10;
+}
+
+type Props = {
+  data: StockChartData[];
+};
+
+const Container = styled.div`
+  height: 32rem;
+  padding: 2rem;
+  margin: 2rem;
+`;
+
+const StockChart: React.FC<Props> = ({ data }) => {
+  const theme = useTheme() as ThemeType;
+  const isAggregate: boolean = data[0].low !== undefined;
+  // debugger;
+  const margin = 0.05;
+  const lowestChartTick =
+    Math.min(...data.map((d) => d.low ?? d.avg)) * (1.0 - margin);
+  const highestChartTick =
+    Math.max(...data.map((d) => d.high ?? d.avg)) * (1.0 + margin);
+
+  const yDomain = [lowestChartTick, highestChartTick];
+
+  const yTicks = linspace(
+    roundTo10(lowestChartTick),
+    roundTo10(highestChartTick),
+    20.0
+  );
+  return (
+    <>
+      <Container>
+        <ResponsiveContainer width="100%" height="100%" debounce={1}>
+          <LineChart data={data} margin={{ bottom: 160, right: 20 }}>
+            <XAxis dataKey="date" angle={-90} textAnchor="end" interval={0} />
+            <YAxis domain={yDomain} ticks={yTicks} />
+            <CartesianGrid stroke="#eee" strokeDasharray="5 5" />
+            <Tooltip />
+            <Legend verticalAlign="top" align="right" />
+
+            {isAggregate && (
+              <Line
+                name="Low price"
+                type="monotone"
+                dataKey="low"
+                stroke={theme.palette.primary.light}
+              />
+            )}
```

Comment:
Zobacz https://react.dev/reference/react/Fragment

```suggestion
            {isAggregate && (
              <>
                  <Line
                    name="Low price"
                    type="monotone"
                    dataKey="low"
                    stroke={theme.palette.primary.light}
                  />
                  <Line
                    name="High price"
                    type="monotone"
                    dataKey="high"
                    stroke={theme.palette.primary.dark}
                  />
              </>
            )}
```